docs/proposals/22-autoscaling-kube-apiserver-via-independent-hpa-and-vpa.md

```diff
@@ -0,0 +1,354 @@
+  - [Alternatives](#alternatives)
+  - [References](#references)
+
+## Summary
+When it comes to autoscaling shoot control plane `kube-apiserver` instances (ShootKapi hereafter), Gardener needs
+both stability and efficiency (accurate scaling). The existing approach of fusing HPA and VPA into the 2-dimensional
+autoscaler which is HVPA, poses severe algorithmic limitations which manifest as stability and efficiency issues in the
+field. 
+
+This document outlines the main deficiencies in the existing solution and proposes a simple design which overcomes them,
+reduces maintenance effort by reducing Gardener's custom code base, and enables subsequent reduction in ShootKapi
+compute cost.
+
+## Motivation
+Gardener uses a relatively complex approach for scaling shoot control plane `kube-apiserver` instances (ShootKapi).
+This need is a consequence of the following concerns:
+1. Gardener supplies K8s clusters of highly diverse compute scale and utilization - ranging from a couple of small
+   practically idle nodes, to hundreds of heavily utilized giants. Furthermore, the ratio of 'cluster compute capacity'
+   to 'ShootKapi load' substantially varies between different applications utilizing Gardener shoots. As a result,
+   compute resources required by ShootKapis of different shoots vary by two orders of magnitude.
+2. The compute cost of ShootKapi workloads constitutes a major part of Gardener's overall compute cost. Small relative
+   waste in sizing ShootKapis results in considerable absolute cost.
+3. Kube-apiserver does not scale well horizontally, in a sense that extra replicas come with substantial compute overhead.   
+
+In short, ShootKapi needs to be scaled:
+- Within a broad range
+- With reasonable accuracy, relative to actual resource consumption
+
+The implication is that ShootKapi requires both horizontal and vertical scaling. The existing implementation utilized 
+to that end is [HVPA].
+
+### Existing Solution
+[HVPA] strives to achieve a fusion of HPA and VPA recommendations:
+`HvpaRecommendedCapacity = k * HpaRecommendedCapacity + (1 - k) * VpaRecommendedCapacity`
+The underlying goal is to allow HPA and VPA to be driven by the same CPU and memory usage metrics, and overcome the
+HPA-VPA incompatibility known to exist in that mode.
+
+### Deficiencies of the Existing Solution
+HVPA strives to achieve a linear combination of HPA's and VPA's scaling policies:<br/>
+`HvpaRecommendedCapacity = k * HpaRecommendedCapacity + (1 - k) * VpaRecommendedCapacity`<br/>
+However, due to technical factors discussed below, it is not possible to fulfil this goal. At different
+points in time, HVPA ends up mixing horizontal and vertical scaling in proportion which arbitrarily varies in the
+entire range between horizontal-only scaling and vertical-only scaling.
+
+#### Algorithmic limitation: fusing the behavior of two blackbox systems 
```

Comment:
I still think this section is way too abstract. I get the point and agree with it, but putting such theoretical work in a proposal document doesn't provide any benefit.
But I can ignore the section for now.

Reply:
I extracted the algorithmic issues section to a separate document and referenced it from the GEP.